### PR TITLE
Add social network requirement to faucet

### PR DESCRIPTION
**Pull request by MakMuftic**
The page is not responsive, fix is needed due to old CSS classes.

### CODE REVIEW

File: src/helpers/tweet-helper.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+const scrape = require('page-scraper');
+const level = require('level');
+const stringSimilarity = require('string-similarity');
+
+// similarity constants
+const referenceString = "Lisinski Faucet Ether Lisinski Testnet PIPA pipa https://lisinski.online/en";
```
Comment (mpetrunic) [inline src/helpers/tweet-helper.js]: This is not same text as tweeted

File: src/helpers/tweet-helper.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+const similarityTreshold = 0.2;
```
Comment (mpetrunic) [inline src/helpers/tweet-helper.js]: If you correct the text yu can probably bump this to 0.8 :)

File: src/helpers/tweet-helper.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+const db = level('tweet-db');
+
+async function checkIfValidTweet (tweetUrl) {
+  const response = {valid: false, message: ""};
+  if (await isNewTweet(tweetUrl)) {
+    const tweetContent = scrapeTweetContent(tweetUrl);
```
Comment (mpetrunic) [inline src/helpers/tweet-helper.js]: you need await here

File: src/helpers/tweet-helper.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    if (tweetContent.length === 0) {
```
Comment (mpetrunic) [inline src/helpers/tweet-helper.js]: !== *

File: src/helpers/tweet-helper.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  }
+}
+
+async function scrapeTweetContent(tweetUrl) {
+  let content = '';
+  try {
+    const $ = await scrape(tweetUrl);
+    content = $('.tweet-text').text();
+  } catch (error) {
+    console.log(error);
```
Comment (mpetrunic) [inline src/helpers/tweet-helper.js]: error.message
It spamms the logs otherwiser

File: src/helpers/tweet-helper.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  }
+  return response;
+}
+
+async function isNewTweet(tweetUrl) {
+  try {
+    await db.get(tweetUrl);
+    return false;
+  } catch (error) {
+    console.log(error);
```
Comment (mpetrunic) [inline src/helpers/tweet-helper.js]: error.message
It spamms the logs otherwise

**Review by mpetrunic — CHANGES_REQUESTED** [review]

File: src/helpers/tweet-helper.js (hunk 5 above)
Comment (Swader) [reply src/helpers/tweet-helper.js]: `console.error()` even

Comment (mpetrunic) [reply src/helpers/tweet-helper.js]: yeah, definitely console.error here

**Comment by MakMuftic** [comment]
@mpetrunic 
- I need to highlight that twitter **doesn't allow to tweet same exact text over and over**, so the current solution in which we are more or less (80% must be same) forcing a user to tweet same text is not ideal but good enough for now.
- Captcha generated element is still breaking page on mobile phones but I will investigate this, and publish some fix in the near future.

**Pull request closed without merging**